python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -358,6 +359,14 @@ def on_http_response_finished(self, reply):
                 # This means that the SSO login worked, but that the user does
                 # have access to the site.
                 session.error = HTTP_CANT_AUTHENTICATE_SSO_NO_ACCESS
+            elif error is QtNetwork.QNetworkReply.NetworkError.AuthenticationRequiredError:
+                # This means that the user entered incorrect credentials.
+                if url.startswith(session.host):
+                    session.error = HTTP_AUTHENTICATE_REQUIRED
+                else:
+                    # If we are on the IdP portal site, we let it deal with the error.
```

Comment:
What is IdP? Could you spell it out in the code?

Reply:
IdP is 'Identity Provider', a very common TLA when the topic of conversation is SSO and/or SAML.